tests/test_pipeline_mixin.py

```diff
@@ -218,27 +271,66 @@ def run_model_pipeline_tests(
         # `run_pipeline_test`.
         pipeline_test_class_name = pipeline_test_mapping[task]["test"].__name__
 
-        for tokenizer_name in tokenizer_names:
-            for processor_name in processor_names:
-                if self.is_pipeline_test_to_skip(
-                    pipeline_test_class_name,
-                    model_architecture.config_class,
-                    model_architecture,
-                    tokenizer_name,
-                    processor_name,
-                ):
-                    logger.warning(
-                        f"{self.__class__.__name__}::test_pipeline_{task.replace('-', '_')}_{torch_dtype} is skipped: test is "
-                        f"currently known to fail for: model `{model_architecture.__name__}` | tokenizer "
-                        f"`{tokenizer_name}` | processor `{processor_name}`."
-                    )
-                    continue
-                self.run_pipeline_test(
-                    task, repo_name, model_architecture, tokenizer_name, processor_name, commit, torch_dtype
+        test_cases = [
+            {
+                "tokenizer_name": tokenizer_name,
+                "image_processor_name": image_processor_name,
+                "feature_extractor_name": feature_extractor_name,
+                "processor_name": processor_name,
+            }
+            for tokenizer_name in tokenizer_names
+            for image_processor_name in image_processor_names
+            for feature_extractor_name in feature_extractor_names
+            for processor_name in processor_names
+        ]
```

Comment:
Note for other reviewers: When a model doesn't support one of these preprocessor types, the list of names will be `[None]`, **not** an empty list. When I first looked at this quadruple-loop I was worried that `test_cases` would be empty if any of the four name lists was empty, but that should never occur. For safety, we might want to add a check for that, though, since it depends on behaviour in a different function that might be modified later!

Reply:
You are right, let's have this check closer to the loop to ensure we always have something to test
https://github.com/huggingface/transformers/pull/32514/commits/6ff4e68ec9caa17f4f5f2535f00f0fbca5fda115

Reply:
After this change, you can also remove

```
            if not isinstance(processor_names, (list, tuple)):
                processor_names = [processor_names]
```
(which was there for the same reason)

Reply:
I left it there to make sure we do not have `str` processor_names